contracts/campaignAirdrops/ExistingERC20DropCampaign.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.15;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../interfaces/AirdropMerkleProof.sol";
+import "../interfaces/IAirBroFactory.sol";
+import "../interfaces/campaignAirdrops/CampaignAirdropsShared.sol";
+
+/// @title Airdrops existing ERC20 tokens for airdrop recipients
+contract ExistingERC20DropCampaign is AirdropMerkleProof, CampaignAidropsShared {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable rewardToken;
+    uint256 public immutable tokenSupply;
+
+    string public constant airdropType = "ERC20";
+    uint256 internal numberOfClaimers;
+    uint256 public tokensPerClaim;
+    uint256 public airdropExpirationTimestamp;
+
+    address internal airdropFunder;
+
+    event MerkleRootSet(bytes32 merkleRoot, uint256 _numberOfClaimers);
+
+    error AlreadyFunded();
+    error AirdropStillActive();
+    error AirdropExpired();
+
+    modifier onlyAdmin() {
+        if (msg.sender != IAirBroFactory(airbroCampaignFactoryAddress).admin()) revert Unauthorized();
+        _;
+    }
+
+    constructor(
+        address _rewardToken,
+        uint256 _tokenSupply,
+        address _airbroCampaignFactoryAddress
+    ) CampaignAidropsShared(_airbroCampaignFactoryAddress) {
+        rewardToken = IERC20(_rewardToken);
+        tokenSupply = _tokenSupply;
+    }
+
+    /// @notice Sets the merkleRoot and the number of claimers (also setting the amount each claimer receivers).
+    /// Can only be done by admin.
+    /// @param _merkleRoot The root hash of the Merle Tree
+    /// @param _numberOfClaimers The number of users eligible to claim
+    function setMerkleRoot(bytes32 _merkleRoot, uint256 _numberOfClaimers) external onlyAdmin {
+        if (merkleRootSet) revert MerkleRootAlreadySet();
+
+        merkleRootSet = true;
+        merkleRoot = _merkleRoot;
+        airdropExpirationTimestamp = block.timestamp + 60 days;
+        numberOfClaimers = _numberOfClaimers;
+        tokensPerClaim = tokenSupply / numberOfClaimers;
```

Comment:
use `_numberOfClaimers` instead of `numberOfClaimers` (cheaper)

Reply:
For line 57 - good catch ! Updated